spec/model/invoice_spec.rb

```diff
@@ -3,78 +3,67 @@
 require_relative "spec_helper"
 
 RSpec.describe Invoice do
-  subject(:invoice) { described_class.new(id: "50d5aae4-311c-843b-b500-77fbc7778050", begin_time: Time.utc(2025, 3), end_time: Time.utc(2025, 4), invoice_number: "2503-4ddfa430e8-0006", created_at: Time.now, content: {"cost" => 10, "subtotal" => 11, "credit" => 1, "discount" => 0, "resources" => [], "billing_info" => {"country" => "NL"}}, status: "unpaid") }
+  subject(:invoice) { described_class.create(project_id: project.id, begin_time: Time.utc(2025, 3), end_time: Time.utc(2025, 4), invoice_number: "2503-4ddfa430e8-0006", created_at: Time.now, content: {"cost" => 10, "subtotal" => 11, "credit" => 1, "discount" => 0, "resources" => [], "billing_info" => {"country" => "NL"}}, status: "unpaid") }
 
   let(:billing_info) { BillingInfo.create(stripe_id: "cs_1234567890") }
   let(:client) { instance_double(Aws::S3::Client) }
+  let(:project) { Project.create(name: "test") }
 
   before do
-    allow(invoice).to receive(:reload)
-    allow(invoice).to receive(:project).and_return(instance_double(Project, path: "/project/p1", accounts: []))
     allow(Config).to receive(:stripe_secret_key).and_return("secret_key")
     allow(Aws::S3::Client).to receive(:new).and_return(client)
   end
 
+  def update_content(new_content)
+    invoice.content.merge!(new_content.transform_keys(&:to_s))
+    invoice.modified!(:content)
+    invoice.save_changes
+  end
+
   describe ".blob_key" do
     it "returns path with below_minimum_threshold group when status is below_minimum_threshold" do
-      invoice.status = "below_minimum_threshold"
+      invoice.update(status: "below_minimum_threshold")
```

Comment:
For this and other changes using `update_content`, you don't need to save to the database unless whatever being tested is reading from the database and not just accessing the model's stored data.

Reply:
We reload the invoice to get its latest status and avoid double charging. This is necessary for tests that call the `.charge` method.

https://github.com/ubicloud/ubicloud/blob/444b833195933590c68a65049d6bc821109b14ed/model/invoice.rb#L71